### PR TITLE
Add revert reasons to proxies

### LINKED ISSUE #490: Add revert messages
See https://github.com/aragon/aragonOS/issues/438

### PR DESCRIPTION
**Pull request by bingen**
Closes #490.

### CODE REVIEW

File: apps/finance/contracts/Finance.sol [hunk 1]
```diff
@@ -32,6 +32,25 @@ contract Finance is EtherTokenConstant, IsContract, AragonApp {
     uint256 internal constant NO_PAYMENT = 0;
     uint256 internal constant NO_TRANSACTION = 0;
 
+    string private constant ERROR_COMPLETE_TRANSITION = "FINANCE_COMPLETE_TRANSITION";
+    string private constant ERROR_NO_PAYMENT = "FINANCE_NO_PAYMENT";
+    string private constant ERROR_NO_TRANSACTION = "FINANCE_NO_TRANSACTION";
+    string private constant ERROR_NO_PERIOD = "FINANCE_NO_PERIOD";
+    string private constant ERROR_VAULT_NOT_CONTRACT = "FINANCE_VAULT_NOT_CONTRACT";
+    string private constant ERROR_INIT_PERIOD_TOO_SHORT = "FINANCE_INIT_PERIOD_TOO_SHORT";
+    string private constant ERROR_SET_PERIOD_TOO_SHORT = "FINANCE_SET_PERIOD_TOO_SHORT";
```
Comment (sohkai) [inline apps/finance/contracts/Finance.sol]: `ERROR_INIT_PERIOD_TOO_SHORT` and `ERROR_SET_PERIOD_TOO_SHORT` could probably be collapsed into the same error.

File: apps/finance/contracts/Finance.sol [hunk 2]
```diff
@@ -32,6 +32,25 @@ contract Finance is EtherTokenConstant, IsContract, AragonApp {
...
+    string private constant ERROR_NEW_PAYMENT_AMOUNT_ZERO = "FINANCE_NEW_PAYMENT_AMOUNT_ZERO";
+    string private constant ERROR_RECOVER_AMOUNT_ZERO = "FINANCE_RECOVER_AMOUNT_ZERO";
+    string private constant ERROR_DEPOSIT_AMOUNT_ZERO = "FINANCE_DEPOSIT_AMOUNT_ZERO";
+    string private constant ERROR_BUDGET = "FINANCE_BUDGET";
+    string private constant ERROR_EXECUTE_PAYMENT_TIME = "FINANCE_EXECUTE_PAYMENT_TIME";
+    string private constant ERROR_RECEIVER_EXECUTE_PAYMENT_TIME = "FINANCE_RCVR_EXEC_PAYMENT_TIME";
```
Comment (sohkai) [inline apps/finance/contracts/Finance.sol]: `ERROR_EXECUTE_PAYMENT_TIME` and `ERROR_RECEIVER_EXECUTE_PAYMENT_TIME` could probably be collapsed into one.

File: apps/vault/contracts/Vault.sol [hunk 3]
```diff
@@ -50,12 +58,12 @@ contract Vault is EtherTokenConstant, AragonApp, DepositableStorage {
         external
         authP(TRANSFER_ROLE, arr(_token, _to, _value))
     {
-        require(_value > 0);
+        require(_value > 0, ERROR_TRANSFER_VALUE_ZERO);
 
         if (_token == ETH) {
             _to.transfer(_value);
```
Comment (sohkai) [inline apps/vault/contracts/Vault.sol]: Wondering if we should use a `require(_to.send(_value))` here so we can customize the error message.

File: apps/vault/contracts/Vault.sol [hunk 4]
```diff
@@ -9,6 +9,14 @@ import "@aragon/os/contracts/lib/token/ERC20.sol";
 contract Vault is EtherTokenConstant, AragonApp, DepositableStorage {
     bytes32 public constant TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
 
+    string private constant ERROR_DATA_NON_ZERO = "VAULT_DATA_NON_ZERO";
+    string private constant ERROR_NOT_DEPOSITABLE = "VAULT_NOT_DEPOSITABLE";
+    string private constant ERROR_DEPOSIT_VALUE_ZERO = "VAULT_DEPOSIT_VALUE_ZERO";
+    string private constant ERROR_TRANSFER_VALUE_ZERO = "VAULT_TRANSFER_VALUE_ZERO";
+    string private constant ERROR_VALUE_MISMATCH = "VAULT_VALUE_MISMATCH";
+    string private constant TOKEN_TRANSFER_FROM_REVERTED = "VAULT_TOKEN_TRANSFER_FROM_REVERT";
+    string private constant TOKEN_TRANSFER_REVERTED = "VAULT_TOKEN_TRANSFER_REVERTED";
```
Comment (sohkai) [inline apps/vault/contracts/Vault.sol]: These last two should have the `ERROR_` prefix :).

File: apps/voting/contracts/Voting.sol [hunk 5]
```diff
@@ -176,7 +186,7 @@ contract Voting is IForwarder, AragonApp {
     * @param _evmScript Start vote with script
     */
     function forward(bytes _evmScript) public {
-        require(canForward(msg.sender, _evmScript));
+        require(canForward(msg.sender, _evmScript), ERROR_CAN_NOT_FORWARD);
         _newVote(_evmScript, "", true, true);
     }
```
Comment (sohkai) [inline apps/voting/contracts/Voting.sol]: Can't comment far enough down, but there should be a message for the `require(vote_.totalVoters > 0)` later.

File: apps/voting/contracts/Voting.sol [hunk 6]
```diff
@@ -162,7 +172,7 @@ contract Voting is IForwarder, AragonApp {
     * @param _voteId Id for vote
     */
     function executeVote(uint256 _voteId) external voteExists(_voteId) {
-        require(canExecute(_voteId));
+        require(canExecute(_voteId), ERROR_CAN_NOT_EXCUTE);
```
Comment (sohkai) [inline apps/voting/contracts/Voting.sol]: **EXECUTE**

File: apps/vault/contracts/Vault.sol (hunk 4 above)
Comment (bingen) [reply apps/vault/contracts/Vault.sol]: I was not using the prefix when it's a `_REVERT`, but I'll add it.

File: apps/voting/contracts/Voting.sol (hunk 5 above)
Comment (bingen) [reply apps/voting/contracts/Voting.sol]: Oh, I missed that one when merging `master`, thanks!

**Pull request merged**